app/screens/edit_post/edit_post.js

```diff
@@ -80,23 +81,39 @@ export default class EditPost extends PureComponent {
         this.focus();
     }
 
-    componentWillReceiveProps(nextProps) {
-        const {editPostRequest} = nextProps;
+    static getDerivedStateFromProps(props, state) {
+        const {status, error} = props.editPostRequest;
+        if (status === RequestStatus.STARTED) {
+            return {
+                ...state,
+                editing: true,
+                error: null,
+            };
+        } else if (status === RequestStatus.FAILURE) {
+            return {
+                ...state,
+                editing: false,
+                error,
+            };
+        }
+
+        return null;
```

Comment:
So I;m going to be a pain in the elbow and ask you to not rely on RequestStatus.. basically forget that we are using the request state as we should remove it completely with the cleanup